editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/maven/MavenUtils.java

```diff
@@ -299,11 +334,28 @@ public static boolean configureCamelVersionForProject(IProject project, String c
 	 * @return	true on success, otherwise false
 	 */
 	public static boolean configurePomCamelVersion(IProject project, CommonNewProjectMetaData projectMetaData, String camelVersion, IProgressMonitor monitor) {
-		String newCamelVersion = Strings.isBlank(camelVersion) && projectMetaData != null && !Strings.isBlank(projectMetaData.getCamelVersion()) ? projectMetaData.getCamelVersion() : camelVersion;
-		SubMonitor subMonitor = SubMonitor.convert(monitor,Messages.mavenTemplateConfiguratorAdaptingprojectToCamelVersionMonitorMessage, 8);
+		SubMonitor subMonitor = SubMonitor.convert(monitor,Messages.mavenTemplateConfiguratorAdaptingprojectToCamelVersionMonitorMessage, 7);
 		try {
 			File pomFile = new File(project.getFile(IMavenConstants.POM_FILE_NAME).getLocation().toOSString()); //$NON-NLS-1$
 			Model m2m = new CamelMavenUtils().getMavenModel(project);
+			
+			boolean ok = configurePomCamelVersion(project, m2m, projectMetaData, camelVersion, subMonitor.newChild(6));
+			if (ok) {
+				new org.fusesource.ide.camel.editor.utils.MavenUtils().writeNewPomFile(project, pomFile, m2m, subMonitor.newChild(1));
+			} else {
+				return false;
+			}
+		} catch (Exception ex) {
+			ProjectTemplatesActivator.pluginLog().logError(ex);
+			return false;
+		}
+		return true;
+	}
+	
+	public static boolean configurePomCamelVersion(IProject project, Model m2m, CommonNewProjectMetaData projectMetaData, String camelVersion, IProgressMonitor monitor) {
+		String newCamelVersion = Strings.isBlank(camelVersion) && projectMetaData != null && !Strings.isBlank(projectMetaData.getCamelVersion()) ? projectMetaData.getCamelVersion() : camelVersion;
+		SubMonitor subMonitor = SubMonitor.convert(monitor, Messages.mavenTemplateConfiguratorAdaptingprojectToCamelVersionMonitorMessage, 8);
+		try {
 			subMonitor.setWorkRemaining(7);
```

Comment:
as extracted in a subclass the setWorkRemaining to 7 becomes useless and the total when creating the submonitor is no more 8

Reply:
what is your suggestion?

Reply:
remove this setWorkremaining and adjust the SubMonitor.convert to match the real number of ticks used in the method